src/easier68k/core/models/assembly_parameter.py

```diff
@@ -0,0 +1,219 @@
+"""
+Assembly Parameter
+
+Represents a single parameter in assembly code
+this is the new 'EAMode' class
+"""
+
+from ..enum.ea_mode import EAMode
+from ..enum.register import Register
+from ...simulator.m68k import M68K
+from ..util.conversions import to_word
+
+# should try to make this a constant only defined once
+MAX_MEMORY_LOCATION = 16777216  # 2^24
+
+
+class AssemblyParameter:
+
+    def __init__(self, mode: EAMode = EAMode.ERR, data: int = 0, length: int = 4):
+        """
+        Constructor
+
+        Checks that the mode and the data are considered to be valid, like
+        registers are within the correct bounds and addresses are not out of bounds.
+        """
+
+
+        # ensure that the values are valid
+
+        # when referencing a register, ensure that the data is within [0, 7]
+        if mode in [EAMode.DataRegisterDirect, EAMode.AddressRegisterIndirectPreDecrement,
+                    EAMode.AddressRegisterIndirect, EAMode.AddressRegisterIndirectPostIncrement,
+                    EAMode.AddressRegisterDirect]:
+            assert 0 <= data <= 7, 'The register number for this mode must be in the range [0, 7]!'
+
+        if mode in [EAMode.AbsoluteWordAddress, EAMode.AbsoluteLongAddress]:
+            # ensure that the address isn't outside the bounds of max memory location
+            # and is greater than or eq to 0
+            assert 0 <= data <= MAX_MEMORY_LOCATION, 'An absolute address must be in the bounds [0, 2^24]!'
+
+        if mode is EAMode.Immediate:
+            # ensure that the value isn't too large, cannot be larger than a long word
+            # negative values need to be converted into unsigned integers
+            assert 0 <= data <= 0xFFFFFFFF, 'Value must fit inside a long word!'
+
+        assert mode is not EAMode.ERR, 'EAMode cannot be ERROR.'
+
+        assert length == 1 or length == 2 or length == 4, 'Length must 1 (byte) 2 (word) 4 (long)'
+
+        # set values
+        self.mode = mode
+        self.data = data
+        self.length = length
+
+    def __str__(self):
+        """
+        str util method
+        :return:
+        """
+        return "EA Mode: {}, Data: {}".format(self.mode, self.data)
+
+    def get_value(self, simulator: M68K) -> int:
+        """
+        Gets the value for this EAMode
+        :param simulator:
+        :return:
```

Comment:
Incomplete docstring; simulator is pretty self explanatory but the return statement might be a bit useful to explain. Also, I feel that the documentation itself could use some work: "gets the value" is a bit vague and I had to actually read into the method to figure out what it was doing.